backend/doi/doi.go

```diff
@@ -0,0 +1,673 @@
+for a running doi backend.
+
+Usage Examples:
+
+    rclone backend set doi: [-o opt_name=opt_value] [-o opt_name2=opt_value2]
+    rclone rc backend/command command=set fs=doi: [-o opt_name=opt_value] [-o opt_name2=opt_value2]
+    rclone rc backend/command command=set fs=doi: -o doi=NEW_DOI
+
+The option keys are named as they are in the config file.
+
+This rebuilds the connection to the doi backend when it is called with
+the new parameters. Only new parameters need be passed as the values
+will default to those currently in use.
+
+It doesn't return anything.
+`,
+}}
+
+// Command the backend to run a named command
+//
+// The command run is name
+// args may be used to read arguments from
+// opts may be used to read optional arguments from
+//
+// The result should be capable of being JSON encoded
+// If it is a string or a []string it will be shown to the user
+// otherwise it will be JSON encoded and shown to the user like that
+func (f *Fs) Command(ctx context.Context, name string, arg []string, opt map[string]string) (out interface{}, err error) {
+	switch name {
+	case "metadata":
+		return f.ShowMetadata(ctx)
+	case "title":
+		return f.ShowTitle(ctx)
+	case "provider":
+		return string(f.provider), nil
+	case "set":
+		newOpt := f.opt
+		err := configstruct.Set(configmap.Simple(opt), &newOpt)
+		if err != nil {
+			return nil, fmt.Errorf("reading config: %w", err)
+		}
+		_, err = f.httpConnection(ctx, &newOpt)
+		if err != nil {
+			return nil, fmt.Errorf("updating session: %w", err)
+		}
+		f.opt = newOpt
+		keys := []string{}
+		for k := range opt {
+			keys = append(keys, k)
+		}
+		fs.Logf(f, "Updated config values: %s", strings.Join(keys, ", "))
+		return nil, nil
+	default:
+		return nil, fs.ErrorCommandNotFound
+	}
+}
+
+// ShowMetadata returns the metadata associated with the DOI
+func (f *Fs) ShowMetadata(ctx context.Context) (metadata interface{}, err error) {
+	metadataURL := f.endpoint
+	var result any
+	opts := rest.Opts{
+		Method:  "GET",
+		RootURL: metadataURL.String(),
+	}
+	err = f.pacer.Call(func() (bool, error) {
+		res, err := f.srv.CallJSON(ctx, &opts, nil, &result)
+		return shouldRetry(ctx, res, err)
+	})
+	if err != nil {
+		return nil, err
+	}
+	return result, nil
+}
+
+// ShowTitle returns the title associated with the DOI
+func (f *Fs) ShowTitle(ctx context.Context) (title string, err error) {
+	switch f.provider {
+	case Dataverse:
+		metadata, err := f.getMetadataDataverse(ctx)
+		if err != nil {
+			return "", err
+		}
+		for _, field := range metadata.Data.LatestVersion.MetadataBlocks.Citation.Fields {
+			if strings.ToLower(field.TypeName) == "title" {
+				title, ok := field.Value.(string)
+				if ok {
+					return title, nil
+				}
+			}
+		}
+	case Invenio, Zenodo:
+		metadata, err := f.getMetadataInvenio(ctx)
+		if err != nil {
+			return "", err
+		}
+		return metadata.Metadata.Title, nil
+
+	}
+	return "<unknown title>", nil
```

Comment:
question: shouldn't you return an error if you cannot find the title?

Reply:
Ok I guess the rclone authors can decide how to address this. If I saw a function and it returned an error and something and if the error is nil then I would expect that that "something" should be normal. In this situation that is not the case.